### PR TITLE
Fix Grafana dashboard

### PR DESCRIPTION
**Pull request by weiiwang01**
### Overview

Fix the Grafana dashboard definition in the charm. Also, update the statsd metrics mapping for better clarity.
 
### Rationale

The Grafana dashboard doesn't function properly with its current configuration.

### Juju Events Changes

None.

### Module Changes

None.

### Library Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -103,7 +104,23 @@ def _on_config_changed(self, _event: ops.EventBase) -> None:
 
     def _on_statsd_prometheus_exporter_pebble_ready(self, _event: ops.PebbleReadyEvent) -> None:
         """Handle the statsd-prometheus-exporter-pebble-ready event."""
-        statsd_container = self.unit.get_container("statsd-prometheus-exporter")
+        container = self.unit.get_container("statsd-prometheus-exporter")
```
Comment (jdkandersson) [inline src/charm.py]: Do yopu think that this should go into the observability observer? Reasoning: this seems to be related to observability and there is an observability observer already that could be a good place for this

**Review by jdkandersson — DISMISSED** [review]

File: src/charm.py (hunk 1 above)
Comment (weiiwang01) [reply src/charm.py]: Gotcha, updated, thanks!

**Review by jdkandersson — APPROVED** [review]

File: src/observability.py [hunk 2]
```diff
@@ -36,3 +40,49 @@ def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
...
+        )
+
+    def _on_statsd_prometheus_exporter_pebble_ready(self, _event: ops.PebbleReadyEvent) -> None:
+        """Handle the statsd-prometheus-exporter-pebble-ready event."""
+        container = self._charm.unit.get_container("statsd-prometheus-exporter")
+        container.push(
+            "/statsd.conf",
+            textwrap.dedent(
+                """\
+                mappings:
```
Comment (amandahla) [inline src/observability.py:55]: I had a review [here](https://github.com/canonical/synapse-operator/pull/30#discussion_r1315856316) regarding harded code configuration. Do you think it applies to this case too?

Comment (weiiwang01) [reply src/observability.py:55]: I believe this is slightly different. It's a static string that is written to the configuration file without any modifications.

**Review by amandahla — APPROVED** [review]
LGTM

**Pull request merged**